### PR TITLE
Development

### PR DESCRIPTION
**Pull request by KostyalBalint**
- Created main readme explaning the architecture, and the development process
- Created docker build, and compose files with support for yarn workspaces
- Env variables are now read with `env-var` package, and a config object is created by workspace
     - This ensures proper type, and default value checks
- Added `typebox` validation (with TS support) for user request validation
- Added TTL for messages placed on the queue, this prevents infinit send loop to occure
- Refactoring
- Request login keys comes from env

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,67 @@
...
+yarn dev
+```
+
+## The architecture
+
+```mermaid
+    flowchart LR
+    A(((Send\nRequest))) --> B[Sender]
+    B --> C[RabitMQ]
+    C --> D[Reciver]
```
Comment (berenteb) [inline README.md]: Typo: Receiver

File: packages/common/index.ts [hunk 2]
```diff
@@ -1,6 +1,24 @@
 export * from "./clientBase";
-export interface Email {
-  subject: string;
-  text: string;
-  to: string;
-}
+export * from "./validateRequest";
+import { Static, TSchema, Type } from "@sinclair/typebox";
```
Comment (berenteb) [inline packages/common/index.ts:3]: Interesting yup alternative

File: packages/consumer/src/mailClient.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  public sendEmail() {
+    if (!this.transporter) {
+      throw new Error("Transporter is not initialized");
+    }
+    return async (email: Email) => {
+      await this.transporter!.sendMail({
+        ...email,
+        from: `${email.from || "Kir-Dev"} <${config.FROM_EMAIL}>`,
```
Comment (berenteb) [inline packages/consumer/src/mailClient.ts:29]: Is this needed explicitly? Doesn't the email client automatically constructs this syntax from the email and sender?

File: packages/sender/src/mw/authMW.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    if (!token) return res.status(401).send("Unauthorized");
+
+    console.info(`Authorizing [${token}]`);
+
+    if (config.LOGIN_TOKENS.includes(token)) {
+      console.info("OK");
+      return next();
+    }
+
+    console.info("Not OK");
```
Comment (berenteb) [inline packages/sender/src/mw/authMW.ts:18]: Me

**Review by berenteb — APPROVED** [review]
Minor things but seems to be working. Approve anyways 👍

File: packages/consumer/src/mailClient.ts (hunk 3 above)
Comment (Tschonti) [reply packages/consumer/src/mailClient.ts:29]: it's important that we can specify the sender. For example, emails sent by Konzisite will show up as if Konzisite sent them and not Kir-Dev

**Review by Tschonti — APPROVED** [review]
Looks good to me, thanks!

File: packages/sender/src/mw/authMW.ts (hunk 4 above)
Comment (KostyalBalint) [reply packages/sender/src/mw/authMW.ts:18]: That was your code, I just didn't touch it xD

File: packages/consumer/src/mailClient.ts (hunk 3 above)
Comment (KostyalBalint) [reply packages/consumer/src/mailClient.ts:29]: We can, that's just a fallback

**Pull request merged**